controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -306,8 +323,19 @@ func (r *StorageConsumerReconciler) reconcileEnabledPhases() (reconcile.Result,
 			r.storageConsumer.Status.State = v1alpha1.StorageConsumerStateReady
 		}
 
-	} else {
+	}
+
+	if markedForDeletion {
 		r.storageConsumer.Status.State = v1alpha1.StorageConsumerStateDeleting
+		if r.storageConsumer.Status.Client == nil || hasForceDeleteAnnotation {
```

Comment:
Why do we care about the client's status at this point?

Reply:
If the status is not nil that means that we have a client onboarded and we shouldn't be deleting the consumer until we have the force delete annotation